server.go

```diff
@@ -404,6 +399,9 @@ func MaxSendMsgSize(m int) ServerOption {
 // MaxConcurrentStreams returns a ServerOption that will apply a limit on the number
 // of concurrent streams to each ServerTransport.
 func MaxConcurrentStreams(n uint32) ServerOption {
+	if n == 0 {
```

Comment:
Also, the same section in the RFC does say that `0` should not be treated as a special value in any sense:

```
      A value of 0 for SETTINGS_MAX_CONCURRENT_STREAMS SHOULD NOT be
      treated as special by endpoints.  A zero value does prevent the
      creation of new streams; however, this can also happen for any
      limit that is exhausted with active streams.  Servers SHOULD only
      set a zero value for short durations; if a server does not wish to
      accept requests, closing the connection is more appropriate.
```

Reply:
This is not the HTTP layer.  This is the gRPC API layer.  An endpoint might set 0 temporarily, but a server using 0 permanently (and we don't support dynamically changing this in gRPC-Go, so this setting _is_ permanent) would be worthless.  Also, this is the current behavior, and I don't want a behavior change for anyone already using 0 to mean "unlimited streams", as that would completely break them.

Note that in most environments that gRPC is used in, the client is trusted to not be malicious.  So I don't think it's necessarily unexpected that some users may leave this unset or set it to zero explicitly to disable the limit.  That said, I wouldn't recommend it, as even a trusted client could go off the rails at some point due to an unforeseen event, and this is one of many knobs we recommend using to protect servers in case of such events.